specification/cognitiveservices/Language.AnalyzeText/models/common.models.tsp

```diff
@@ -0,0 +1,987 @@
+  @doc("(NOTE: Recommended to use summaryLength over sentenceCount) Controls the approximate length of the output summaries.")
+  summaryLength?: SummaryLengthBucket;
+}
+
+@doc("The context of the summary.")
+model SummaryContext {
+  @doc("Start position for the context. Use of different 'stringIndexType' values can affect the offset returned.")
+  offset: int32;
+
+  @doc("The length of the context. Use of different 'stringIndexType' values can affect the length returned.")
+  length: int32;
+}
+
+@clientName("CustomTaskContent", "csharp")
+@doc("Parameters object for a text analysis task using custom models.")
+model CustomTaskParameters is TaskParameters {
+  @doc("This field indicates the project name for the model.")
+  projectName: string;
+
+  @doc("This field indicates the deployment name for the model.")
+  deploymentName: string;
+}
+
+@doc("Custom Task Results")
+model CustomResult {
+  @doc("Errors by document id.")
+  errors: Array<DocumentError>;
+
+  @doc("if showStats=true was specified in the request this field will contain information about the request payload.")
+  statistics?: RequestStatistics;
+
+  @doc("This field indicates the project name for the model.")
+  projectName: string;
+
+  @doc("This field indicates the deployment name for the model.")
+  deploymentName: string;
+}
+
+@doc("Optional parameter to use a Custom Character to be used for redaction in PII responses. Default character will be * as before. We allow specific ascii characters for redaction.")
+union redactionCharacter {
+  string,
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  exclamationPoint: "!",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  numberSign: "#",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  dollar: "$",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  perCent: "%",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  ampersand: "&",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  asterisk: "*",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  plus: "+",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  minus: "-",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  equals: "=",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  questionMark: "?",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  atSign: "@",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  caret: "^",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  underscore: "_",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  tilde: "~",
+}
+
+@clientName("AnalyzeTextError", "csharp")
+@doc("The error response object returned when the service encounters some errors during processing the request.")
+model Error {
```

Comment:
Add an `@error` decorator.

Reply:
@clientName decorators should go in client.tsp

Reply:
TODO send example of client.tsp

Reply:
Here are some docs on how to do client renaming which should go in client.tsp: https://azure.github.io/typespec-azure/docs/next/howtos/DataPlane%20Generation%20-%20DPG/04renaming#customizations

Reply:
moved to client.tsp